### PR TITLE
NAS-109861 / 21.04 / Selectively add locked key

### PR DESCRIPTION
**Pull request by sonicaj**
This commit introduces a change where we only add `locked` key to `pool.dataset.query` output if encryption properties are retrieved.

### CODE REVIEW

**Comment by themylogin** [comment]
What if we don't want any encryption information, why are we still forced to retrieve that?

**Review by themylogin — APPROVED** [review]

File: src/middlewared/middlewared/plugins/pool.py [hunk 1]
```diff
@@ -2805,6 +2805,10 @@ def query(self, filters, options):
         extra = copy.deepcopy(options.get('extra', {}))
         retrieve_children = extra.get('retrieve_children', True)
         props = extra.get('properties')
+        if props is not None:
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/pool.py:2808]: This looks odd. Why is this needed?

**Review by william-gr — APPROVED** [review]
